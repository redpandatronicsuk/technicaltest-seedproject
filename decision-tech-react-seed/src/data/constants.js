// By default no filter is selected
const defaultProductTypeFilterSelections = {
  Broadband: false,
  TV: false,
  Mobile: false
}

export {
    defaultProductTypeFilterSelections
}