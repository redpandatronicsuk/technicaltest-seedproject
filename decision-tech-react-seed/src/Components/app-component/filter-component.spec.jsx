import 'isomorphic-fetch'
var fetchMock = require('fetch-mock')
import React from 'react'
import Filter from './filter-component'
import renderer from 'react-test-renderer'
import { defaultProductTypeFilterSelections } from '../../data'

test('Snapshot test for filter component', () => {

fetch.mockResponse(JSON.stringify({deals: [{name: 'x'}] }))

  const component = renderer.create(
    <Filter selectedBroadbandFilterTypes={defaultProductTypeFilterSelections}/>
  );
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
});