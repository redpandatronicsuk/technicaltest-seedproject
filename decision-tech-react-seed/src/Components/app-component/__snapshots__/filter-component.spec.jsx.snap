// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test for filter component 1`] = `
<div>
  <ul>
    <li>
      <input
        checked={false}
        onChange={[Function]}
        type="checkbox"
      />
       
      Broadband
    </li>
    <li>
      <input
        checked={false}
        onChange={[Function]}
        type="checkbox"
      />
       
      TV
    </li>
    <li>
      <input
        checked={false}
        onChange={[Function]}
        type="checkbox"
      />
       
      Mobile
    </li>
  </ul>
  <div>
    Speed
  </div>
  <select
    onChange={[Function]}
    value={undefined}
  >
    <option
      value="ANY"
    >
      ANY
    </option>
    <option
      value={76}
    >
      76
    </option>
    <option
      value={52}
    >
      52
    </option>
    <option
      value={17}
    >
      17
    </option>
  </select>
</div>
`;
